### PR TITLE
babel-plugin-macros: Export all types and fix References type

### PR DESCRIPTION
**Pull request by coffee-cup**
This PR does 2 things

- Export all types
- Fix `references` type. Previously it was an object of `Babel.NodePath`. However, [the docs](https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/author.md#references) say it is an object.

I've been unable to lint these packages locally because of the following error

image

Any advice on how to fix it is appreciated.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/kentcdodds/babel-plugin-macros/blob/master/other/docs/author.md#references
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/babel-plugin-macros/index.d.ts [hunk 1]
```diff
@@ -1,36 +1,27 @@
-// Type definitions for babel-plugin-macros 2.6
+// Type definitions for babel-plugin-macros 2.8
 // Project: https://github.com/kentcdodds/babel-plugin-macros
 // Definitions by: Billy Kwok <https://github.com/billykwok>
+//                 Jake Runzer <https://github.com/coffee-cup>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.9
+// TypeScript Version: 3.7
```
Comment (peterblazejewicz) [inline types/babel-plugin-macros/index.d.ts]: Is that intended? `@babel/core` works with 2.9:
```ts
// TypeScript Version: 2.9
```
and your change pass linter with that verison as well.

**Comment by peterblazejewicz** [comment]
> Any advice on how to fix it is appreciated.

So having these errors:
```bash
Error: Errors in typescript@next for external dependencies:
../babel__core/index.d.ts(13,20): error TS2307: Cannot find module '@babel/types'.
../babel__core/index.d.ts(14,31): error TS2307: Cannot find module '@babel/parser'.
../babel__generator/index.d.ts(11,20): error TS2307: Cannot find module '@babel/types'.
../babel__template/index.d.ts(9,31): error TS2307: Cannot find module '@babel/parser'.
../babel__template/index.d.ts(10,54): error TS2307: Cannot find module '@babel/types'.
../babel__traverse/index.d.ts(11,20): error TS2307: Cannot find module '@babel/types'.
```
means: each of those types depends on `@babel/...` dependencies. And because these deps are external (not found in the types/... directory), you have to:
- go to each of the directovies listed (`types/babel__core`, etc)
- run `npm install`
After that, rerun your test and it shoudl work

File: types/babel-plugin-macros/index.d.ts (hunk 1 above)
Comment (coffee-cup) [reply types/babel-plugin-macros/index.d.ts]: oops, not intended. I'll revert.

**Comment by coffee-cup** [comment]
@peterblazejewicz thanks for the advice. Linting works now

**Comment by peterblazejewicz** [comment]
/cc @billykwok 
I'm not familiar with this lib, can you look? Thx!

**Comment by billykwok** [comment]
The change looks good to me.
Thanks @coffee-cup for correcting the `references` type.

**Review by billykwok — APPROVED** [review]
Looks good. I wasn't aware that `references` should be an object when I first wrote the definition. Exporting related types would be helpful too.
Thanks @coffee-cup

**Pull request merged**